include/crocoddyl/multibody/costs/control-gravity-contact.hxx

```diff
@@ -0,0 +1,93 @@
+    : Base(state, activation, nu), pin_model_(state->get_pinocchio()) {
+  if (activation_->get_nr() != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "nr is equals to " + std::to_string(nu_));
+  }
+}
+
+template <typename Scalar>
+CostModelControlGravContactTpl<Scalar>::CostModelControlGravContactTpl(boost::shared_ptr<StateMultibody> state)
+    : Base(state, state->get_nv()), pin_model_(state->get_pinocchio()) {
+}
+
+template <typename Scalar>
+CostModelControlGravContactTpl<Scalar>::CostModelControlGravContactTpl(boost::shared_ptr<StateMultibody> state,
+                                                 const std::size_t& nu)
+    : Base(state, nu, nu), pin_model_(state->get_pinocchio()) {
+}
+
+template <typename Scalar>
+CostModelControlGravContactTpl<Scalar>::~CostModelControlGravContactTpl() {}
+
+template <typename Scalar>
+void CostModelControlGravContactTpl<Scalar>::calc(const boost::shared_ptr<CostDataAbstract>& data,
+                                       const Eigen::Ref<const VectorXs>& x, const Eigen::Ref<const VectorXs>& u) {
+  if (nu_ == 0) {
+    throw_pretty("Invalid argument: "
+                 << "it seems to be an autonomous system, if so, don't add this cost function");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+  Data* d = static_cast<Data*>(data.get());
+
+  data->r = u - pinocchio::rnea(*pin_model_,*(d->pinocchio),x.head(state_->get_nq()),x.tail(state_->get_nv()),Eigen::VectorXd::Zero(state_->get_nv()),d->fext).tail(nu_);
+  activation_->calc(data->activation, data->r);
+  data->cost = data->activation->a_value;
+}
+
+template <typename Scalar>
+void CostModelControlGravContactTpl<Scalar>::calcDiff(const boost::shared_ptr<CostDataAbstract>& data,
+                                           const Eigen::Ref<const VectorXs>& x, const Eigen::Ref<const VectorXs>& u) {
+
+  if (nu_ == 0) {
+    throw_pretty("Invalid argument: "
+                 << "it seems to be an autonomous system, if so, don't add this cost function");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+  Data* d = static_cast<Data*>(data.get());
+
+  pinocchio::computeRNEADerivatives(*pin_model_,*(d->pinocchio),x.head(state_->get_nq()),x.tail(state_->get_nv()),Eigen::VectorXd::Zero(state_->get_nv()),d->fext,
+                                    d->rnea_partial_dx.topRows(state_->get_nv()),d->rnea_partial_dx.bottomRows(state_->get_nv()),d->rnea_partial_da);
+  
+  activation_->calcDiff(data->activation, data->r);
+
+  data->Lu = data->activation->Ar;
+  data->Lx = - d->rnea_partial_dx * data->activation->Ar;
+  data->Lxx = d->rnea_partial_dx * data->activation->Arr * d->rnea_partial_dx.transpose();
```

Comment:
Split expression as described below.
Also do not forget the noalias in the next line